### PR TITLE
Add listener to refresh access token when expired

### PR DESCRIPTION
**Pull request by monteiro**
- Added symfony listener to listen when a response arrives, refreshes token if needed.

### CODE REVIEW

File: Security/Authentication/GoogleApiAuthentication.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    public function __construct(
+        $firewallName,
+        GoogleResourceOwner
+        $googleResourceOwner,
```
Comment (xipasduarte) [inline Security/Authentication/GoogleApiAuthentication.php]: Why is this not in one line? `GoogleResourceOwner $googleResourceOwner`

File: Security/Authentication/GoogleApiAuthentication.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+     * @param  Request $request http request
+     * @return bool    true if the token was refreshed, false otherwise
+     */
+    public function refreshToken(Request $request)
+    {
+        $session = $request->getSession();
+        /**
+         * The name of the session depends of HWIOAuthBundle
+         */
+        $oAuthToken = unserialize($session->get('_security_'.$this->firewallName));
```
Comment (xipasduarte) [inline Security/Authentication/GoogleApiAuthentication.php]: Use spaces before and after concatenation `'_security_' . $this`

File: Tests/Security/Authentication/GoogleApiAuthenticationTest.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @var string
+     */
+    private $firewallName;
+
+    public function setUp()
```
Comment (xipasduarte) [inline Tests/Security/Authentication/GoogleApiAuthenticationTest.php]: What is the difference between this and a `__construct()`?

Comment (monteiro) [inline Tests/Security/Authentication/GoogleApiAuthenticationTest.php]: I'm extending \PHPUnit_Framework_TestCase which will execute setUp() each time it runs a test (it is a public method that has a "test" prefix in this name).

If I just wanted one instance of something I could use __construct. Each test in this case is isolated from the others.

File: Tests/Security/Authentication/GoogleApiAuthenticationTest.php [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    /**
+     * Test when the access token is expired, but there is no token available in the
+     * token storage to be updated.
+     */
+    public function testRefreshTokenExpiredWithoutTokenInStorage()
+    {
+        $today = new \DateTime();
+        $oauthToken = $this->getOAuthToken();
+        $oauthToken->setCreatedAt($today->getTimestamp());
+        $oauthToken->setExpiresIn(-100);
```
Comment (xipasduarte) [inline Tests/Security/Authentication/GoogleApiAuthenticationTest.php]: Take this value out and create a global setting/variable. State the units on the value declaration.

**Pull request merged**